### ISSUE TITLE
Storage and registry glusterfs clusters share the same heketi secret

### ISSUE THREAD

**Issue by mandre** [openshift/openshift-ansible#10712]
#### Description

Deploying separate glusterfs clusters for storage and registry, they both share the same heketi secret:

```
[heat-admin@openshift-openshiftmaster-0 ~]$ sudo oc get secrets --all-namespaces | grep heketi
default                 heketi-registry-config-secret                              Opaque                                2         18m
default                 heketi-registry-service-account-dockercfg-5wdhm            kubernetes.io/dockercfg               1         18m
default                 heketi-registry-service-account-token-2czsq                kubernetes.io/service-account-token   4         18m
default                 heketi-registry-service-account-token-pvbj9                kubernetes.io/service-account-token   4         18m
glusterfs               heketi-storage-admin-secret                                kubernetes.io/glusterfs               1         20m
glusterfs               heketi-storage-config-secret                               Opaque                                2         22m
glusterfs               heketi-storage-service-account-dockercfg-jsqpd             kubernetes.io/dockercfg               1         22m
glusterfs               heketi-storage-service-account-token-6lpdn                 kubernetes.io/service-account-token   4         22m
glusterfs               heketi-storage-service-account-token-p6429                 kubernetes.io/service-account-token   4         22m
```

This causes issue as the scale up playbook is currently unable to retrieve the heketi secret, and even if it was the case as fixed in #10710, the code looks for the "heketi-{{ glusterfs_name }}-admin-secret" secret with a different value for `glusterfs_name`.

During initial deployment of worker and infra nodes, the `glusterfs_name` variable is set to "storage" by default, while it is set to "registry" when performing the infra node scale up.
 
##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

```
ansible 2.6.7
  config file = /usr/share/ansible/openshift-ansible/ansible.cfg
  configured module search path = [u'/opt/app-root/src/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

* Openshift-ansible version:

```
openshift-ansible-3.11.44-1.git.0.11d174e.el7.noarch
```

##### Expected Results

The heketi secret for the registry should be stored in a `heketi-registry-admin-secret` in the `default` namespace.

##### Observed Results
Describe what is actually happening.

The heketi secret for the registry should is stored in a `heketi-storage-admin-secret` in the `glusterfs` namespace. Is it shared with the storage glusterfs cluster.

##### Additional Information

[inventory.tar.gz](https://github.com/openshift/openshift-ansible/files/2590585/inventory.tar.gz)
[playbook.log.gz](https://github.com/openshift/openshift-ansible/files/2590586/playbook.log.gz)

**Comment by mandre** [comment]
We're hitting this issue when scaling out OpenShift from TripleO [1] so decided to take a stab at fixing it.

There is a guard in glusterfs_registry.yml [2] that prevents running the tasks from glusterfs_common.yml when there's also a glusterfs group, presumably because they're already performed as part of the glusterfs_config.yml setup. This results in `glusterfs_heketi_client` variable being the same for the storage and the registry.

On the scale out operation however, there is no such guard and the `glusterfs_heketi_client` is being redefined with the `glusterfs_name` from glusterfs_registry_facts.yml and fails talking to heketi.

I assume that all the `openshift_storage_glusterfs_registry_heketi_*` variables serve no real purpose since they're ignored on initial deployment (and mess up with the scale out) and should default to their non "registry" equivalent.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1645656
[2] https://github.com/openshift/openshift-ansible/blob/aa15bf2/roles/openshift_storage_glusterfs/tasks/glusterfs_registry.yml#L6-L7

**Comment by mandre** [comment]
On second thoughts, I believe skipping the heketi initialization for the registry as it is done in [1] is a bug. The update_topology.yml taskfile used for scale out doesn't have any of these assumption [2] and then deploys the heketi pods for the registry.

The fact that scale out completes when setting `openshift_storage_glusterfs_registry_heketi_admin_key` to the value of heketi-storage-admin-secret is just plain luck.

[1] https://github.com/openshift/openshift-ansible/blob/aa15bf2/roles/openshift_storage_glusterfs/tasks/glusterfs_registry.yml#L6-L7
[2] https://github.com/openshift/openshift-ansible/blob/aa15bf26e982dada288d36fca8d3d9da8a29f31f/roles/openshift_storage_glusterfs/tasks/update_topology.yml#L21-L26

**Comment by mandre** [comment]
cc @michaelgugino

**Comment by jarrpa** [comment]
I believe this is resolved with #11072. Closing this Issue.
